### PR TITLE
4358 Janis Search: Pagination

### PR DESCRIPTION
**Pull request by robbobfrh84**
ZenHub Issues:
- Search Pagination: https://app.zenhub.com/workspaces/techstack-5a78b88e1ce69f3510b678ef/issues/cityofaustin/techstack/4358
- Search Navigation:https://app.zenhub.com/workspaces/techstack-5a78b88e1ce69f3510b678ef/issues/cityofaustin/techstack/4283

Zenhub Issue: https://app.zenhub.com/workspaces/techstack-5a78b88e1ce69f3510b678ef/issues/cityofaustin/techstack/4358

Deployed: https://janis-v3-search-pag.netlify.com/

# Description

This PR will 
- Add Pagination to search
  - Also cleaned up the reusability of the component. There are some of @briaguya event page. changes in here as well. 
- Add hash query helper functions for cross-page uses.
  - I left notes in the code about use-cases. And, a bit of a breakdown of what's going on.
  - Let me know if i can add or clarify in anyway.
- Add Search icon to navigation

# Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
  - Back from pagination was blocking history. That's fixed now
- [x] New feature (non-breaking change which adds functionality)

# How can this be tested?

👇 https://janis-v3-search-pag.netlify.com/
- hit the search button 
- search something, hit the next page on pagination
- go back in pagination.
- search "official"
- Click the only result. 
- paginate some more
- go back. 

NOTE: You may notice having to press back twice after returning back through the pagination. This is because `/search` registers to history then `/search#page=1`. I had some ideas to fix, but felt it was ok for MVP and maybe not even that "wrong" in what you'd expect. 

# Checklist:
- [x] Request reviewers
- [ ] Slack-out message for review
- [ ] Link this PR in the issue
- [ ] Moved card to *review* in zenhub

### CODE REVIEW

File: src/components/PageSections/ContextualNav/index.js [hunk 1]
```diff
@@ -23,7 +23,10 @@ const ContextualNav = ({ parent, relatedTo, offeredBy, intl }) => (
               {`${intl.formatMessage(i18n.relatedTo)}: `}
             </span>
             {relatedTo.map((relatedLinkData, index) => (
-              <Link to={`/${intl.locale}${relatedLinkData.url}`} key={relatedLinkData.url}>
+              <Link
```
Comment (briaguya) [inline src/components/PageSections/ContextualNav/index.js:26]: made an issue for prettier stuff https://github.com/cityofaustin/techstack/issues/4460

Comment (Niicck) [reply src/components/PageSections/ContextualNav/index.js:26]: Maybe let's start holding off on prettier until it's part of the CI flow for every PR.

File: src/components/PageSections/Pagination/paginationContainer.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+              {pages &&
+                shownPages.map((page, i) => (
+                  <PageNumber
+                    pageNumber={pageNumber + 1}
+                    index={shownPages[i]}
+                    paginationIndex={i}
+                    key={i}
+                    pageNumberIndex={shownPages.indexOf(pageNumber + 1)}
+                    changePage={changePage}
+                    contentType={'OfficialDocumentPage'}
```
Comment (Niicck) [inline src/components/PageSections/Pagination/paginationContainer.js:132]: Why is 'OfficialDocumentPage' and `coa-OfficialDocumentPage_page` css classes hardcoded into the paginationContainer?

Comment (robbobfrh84) [reply src/components/PageSections/Pagination/paginationContainer.js:132]: Oh no. They should not. refactor stowaways. I'll clean this up.

File: src/components/PageSections/ContextualNav/index.js (hunk 1 above)
Comment (robbobfrh84) [reply src/components/PageSections/ContextualNav/index.js:26]: @Niicck should I revert this back to the old way then for the time being?

Comment (Niicck) [reply src/components/PageSections/ContextualNav/index.js:26]: nah, that sounds like a pain. Just going forward from now.

**Pull request merged**